lua/nvim-treesitter/indent.lua

```diff
@@ -0,0 +1,96 @@
+local parsers = require'nvim-treesitter.parsers'
+local queries = require'nvim-treesitter.query'
+
+local M = {}
+
+local function get_node_at_line(root, lnum)
+  for node in root:iter_children() do
+    local srow, _, erow = node:range()
+    if srow == lnum then return node end
+
+    if node:child_count() > 0 and srow < lnum and lnum <= erow then
+      return get_node_at_line(node, lnum)
+    end
+  end
+
+  local wrapper = root:descendant_for_range(lnum, 0, lnum, -1)
+  local child = wrapper:child(0)
+  return child or wrapper
+end
+
+local function update_indents(bufnr)
+  local indents = queries.get_capture_matches(bufnr, '@indent', 'indents') or {}
+  local branches = queries.get_capture_matches(bufnr, '@branch', 'indents') or {}
```

Comment:
`queries.get_capture_matches(bufnr, '@indent.node', 'indents')`? then you don't need the subsequent loops.

Reply:
fastest option is of course to use the nvim's upstream iter_queries function and directly look for up matches of `@indent`